lib/internal/Magento/Framework/Image/Adapter/ImageMagick.php

```diff
@@ -562,4 +580,42 @@ private function addSingleWatermark($positionX, int $positionY, \Imagick $waterm
             $compositeChannels
         );
     }
+
+    /**
+     * Get and store the image colorspace.
+     *
+     * @return int
+     */
+    public function getColorspace(): int
+    {
+        if ($this->colorspace === -1) {
+            $this->originalColorspace = $this->colorspace = $this->_imageHandler->getImageColorspace();
+        }
+
+        return $this->colorspace;
+    }
+
+    /**
+     * Get the original image colorspace.
+     *
+     * @return int
+     */
+    public function getOriginalColorspace(): int
```

Comment:
Please, make method private

Reply:
@sidolov I can do that, but then this behavior cannot be tested like it is right now. It depends on these functions being public.

Reply:
This method was need for testing, but I deleted now in favor of requested changes.